src/dust_rng.cpp

```diff
@@ -433,6 +433,46 @@ cpp11::sexp dust_rng_hypergeometric(SEXP ptr, int n,
   return sexp_matrix(ret, n, n_streams);
 }
 
+
+template <typename real_type, typename T>
+cpp11::sexp dust_rng_gamma(SEXP ptr, int n,
+                           cpp11::doubles r_a, cpp11::doubles r_b, int n_threads) {
```

Comment:
for consistency, probably also want this to be shape, scale, though it will make it even nastier. Not sure how we can generally keep this from getting so ugly while still keeping it flexible (it's mostly just for testing)